### PR TITLE
Adjust layout xml instructions and adding xml examples for all argument types

### PR DESCRIPTION
**Pull request by serhiyzhovnir**
## Purpose of this PR

This PR adjust the layout XML instructions and it adds the examples for all layout argument types.

## Affected URLs

- https://devdocs.magento.com/guides/v2.2/frontend-dev-guide/layouts/xml-instructions.html
- https://devdocs.magento.com/guides/v2.3/frontend-dev-guide/layouts/xml-instructions.html

## Links to source code

- https://github.com/magento/magento2/blob/2.3-develop/app/etc/di.xml#L480 - layout arguments interpreters
- https://github.com/magento/magento2/blob/2.2/lib/internal/Magento/Framework/View/Layout/etc/elements.xsd#L375
- https://github.com/magento/magento2/blob/0c786907ffe03d0e2990612eec16ee58b00379c5/lib/internal/Magento/Framework/View/Layout/etc/elements.xsd#L345
- https://github.com/magento/magento2/blob/0c786907ffe03d0e2990612eec16ee58b00379c5/lib/internal/Magento/Framework/View/Layout/etc/elements.xsd#L337

whatsnew
Added Argument type examples in [Layout instructions](https://devdocs.magento.com/guides/v2.3/frontend-dev-guide/layouts/xml-instructions.html).

### CODE REVIEW

**Comment by devops-devdocs** [comment]
An admin must run tests on this PR before it can be merged.

File: guides/v2.2/frontend-dev-guide/layouts/xml-instructions.md [hunk 1]
```diff
@@ -244,7 +244,7 @@ Used to pass an argument. Must be always enclosed in [`<arguments>`](#arguments)
 | Attribute | Description | Values | Required? |
 |:------- |:------ |:------ |:------ |
 | `name` | Argument name. | unique | yes |
-| `xsi:type` | Argument type. | `string|boolean|object|number|null|array` | yes |
+| `xsi:type` | Argument type. | `string|boolean|object|number|null|array|options|url|helper` | yes |
```
Comment (rogyar) [inline guides/v2.2/frontend-dev-guide/layouts/xml-instructions.md]: Codebase references:

https://github.com/magento/magento2/blob/2.2/lib/internal/Magento/Framework/View/Layout/etc/elements.xsd#L375
https://github.com/magento/magento2/blob/0c786907ffe03d0e2990612eec16ee58b00379c5/lib/internal/Magento/Framework/View/Layout/etc/elements.xsd#L345
https://github.com/magento/magento2/blob/0c786907ffe03d0e2990612eec16ee58b00379c5/lib/internal/Magento/Framework/View/Layout/etc/elements.xsd#L337

**Comment by jeff-matthews** [comment]
@magento/devdocs-admins: this still needs an editorial review

**Comment by dobooth** [comment]
Please put codebase references in the PR description.

**Comment by serhiyzhovnir** [comment]
@dobooth code references were added to the PR description.
Thank you!

**Comment by dobooth** [comment]
running tests

**Pull request merged**